astropy/units/quantity.py

```diff
@@ -275,6 +275,14 @@ def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
     def __array_finalize__(self, obj):
         self._unit = getattr(obj, '_unit', None)
 
+        # Copy info (except for a scalar output for performance reasons) if the
```

Comment:
The comment still mentions scalar, which is out of date.

Aside, I agree that this is better than my fastest solution (below), because in principle a class could have its unit only defined in `info`, which would now get passed on.

```
try:
    self._unit = obj._unit
except AttributeError:
    pass
else:
    if 'info' in obj.__dict__:
        self.info = obj.info
```